docs/how-to/add-custom-code.md

```diff
@@ -48,31 +48,39 @@ src
     └── ...
 ```
 
-Amplication may still update the non-base files that include your custom code in certain circumstances. These updates are limited to changes necessary for preventing errors and ensuring the project can build correctly. For example, if you remove a plugin that was previously used in your service, Amplication might update the non-base files to remove references to the removed plugin, thus preventing build errors.
-
-This approach allows Amplication to maintain the integrity of your project structure while still preserving your custom code to the greatest extent possible.
+## Smart Git Sync
 
-## Amplication's Approach to Custom Code
+Amplication uses [Smart Git Sync](/smart-git-sync/) to merge changes, preserving your custom code while updating generated parts. This feature:
 
-Amplication is designed to preserve your custom code while allowing for continuous updates to the generated code. Here's how it works:
+1. Regenerates base files with each build.
+2. Preserves non-base files containing your custom code.
+3. Makes necessary updates to non-base files (e.g., removing references to deleted plugins) while maintaining your custom code.
 
-1. **Initial Generation**: Amplication generates both base and customizable files for each module.
+## Best Practices for Custom Code
 
-2. **Subsequent Updates**: When you make changes in Amplication (e.g., updating the data model), it regenerates the base files and updates the customizable files as needed.
+1. Add custom code to non-base files (e.g., `user.service.ts` instead of `user.service.base.ts`).
+2. Use types and interfaces generated by Amplication to ensure type safety.
+3. Leverage existing services and utilities provided by Amplication.
+4. Document your custom code thoroughly.
+5. Create a new branch for significant custom code changes.
+6. Regularly pull and merge the latest Amplication-generated code into your working branch.
 
-3. **Smart Merging**: Amplication uses [Smart Git Sync](/smart-git-sync/) to merge changes, preserving your custom code while updating the generated parts.
+## Handling Conflicts
 
-4. **Conflict Resolution**: If conflicts arise, Amplication provides clear indications and allows you to resolve them manually.
+While Amplication strives to preserve your custom code, conflicts may arise, especially with significant changes to your data model or entity structure. If conflicts occur:
```

Comment:
Let's emphasize that the conflict resolution should take place on 'amplication' branch, and then- merge the result to the main branch